object.go

```diff
@@ -140,6 +142,43 @@ func (o *Object) DecCyclicalCount() {
 	o.cyclicalCount--
 }
 
+func (o *Object) AddSeenObject(ptr *Object) {
```

Comment:
these new receivers you created don't need to be public, let's lowercase them to make them private functions